programs/escrow/src/lib.rs

```diff
@@ -51,33 +74,26 @@ pub mod escrow {
     /// - transfer funds back to giver
     /// - close the vault_token_account and escrow_account
     pub fn cancel(ctx: Context<Cancel>) -> ProgramResult {
-        let Cancel { giver, giver_token_account, vault_token_account, vault_authority, token_program, escrow_account, .. } = ctx.accounts;
-        if escrow_account.is_released {
-            return Err(ErrorCode::CancelReleasedAccountError.into())
+        if ctx.accounts.escrow_account.is_released {
+            return Err(ErrorCode::CancelReleasedAccountError.into());
         }
 
-        let authority_seeds = &[&ESCROW_PDA_SEED[..], &[escrow_account.vault_authority_bump]];
+        let authority_seeds = &[
+            &ESCROW_PDA_SEED[..],
+            &[ctx.accounts.escrow_account.vault_authority_bump],
+        ];
 
         token::transfer(
-            CpiContext::new(
-                token_program.clone(),
-                Transfer {
-                    from:      vault_token_account.to_account_info().clone(),
-                    to:        giver_token_account.to_account_info().clone(),
-                    authority: vault_authority.clone(),
-                }
-            ).with_signer(&[&authority_seeds[..]]),  // needed as vault_pda wasn't used as signer
-            escrow_account.amount
+            ctx.accounts
+                .into_transfer_context() // Nice way to abstract out some boilerplate logic outside of the business logic.
```

Comment:
to be honest I find that confusing, as eg. it doesn't spell out what kind of transfer. In the anchor escrow demo, I saw this implemented with generic into(), which seemed even less obvious. But if that's the norm, will conform